core/src/test/java/hudson/tasks/BuildStepCompatibilityLayerTest.java

```diff
@@ -16,6 +16,7 @@
 
     @Issue("JENKINS-18734")
     @Test(expected = AbstractMethodError.class)
+    @SuppressWarnings("deprecation")
```

Comment:
This is an example of an appropriate suppression: we are deliberately testing the deprecated variant. Might be worth including a comment on the annotation line reminding readers of that.